src/librustdoc/visit_ast.rs

```diff
@@ -29,6 +29,16 @@ fn def_id_to_path(tcx: TyCtxt<'_>, did: DefId) -> Vec<String> {
     std::iter::once(crate_name).chain(relative).collect()
 }
 
+crate fn inherits_doc_hidden(cx: &core::DocContext<'_>, mut node: hir::HirId) -> bool {
+    while let Some(id) = cx.tcx.hir().get_enclosing_scope(node) {
+        node = id;
+        if cx.tcx.hir().attrs(node).lists(sym::doc).has_word(sym::hidden) {
```

Comment:
nit: it would be nice to only take `tcx`, not `cx`, since the full DocContext isn't used.
```suggestion
crate fn inherits_doc_hidden(tcx: TyCtxt<'_>, mut node: hir::HirId) -> bool {
    while let Some(id) = tcx.hir().get_enclosing_scope(node) {
        node = id;
        if tcx.hir().attrs(node).lists(sym::doc).has_word(sym::hidden) {
```